hail/src/main/scala/is/hail/expr/ir/Emit.scala

```diff
@@ -864,117 +864,55 @@ private class Emit(
           Code._empty)
 
       case ArrayAgg(a, name, query) =>
+        assert(!ContainsAggIntermediate(query))
```

Comment:
added this check while debugging a failure from new Extract where we were forwarding a ResultOp2 inside an ArrayAgg and returning invalid results.